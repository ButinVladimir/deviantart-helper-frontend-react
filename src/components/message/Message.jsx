import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

export default function Message({
  showMessage,
  message,
  hideMessageHandler,
}) {
  return (
    <Fragment>
      {showMessage && (
        <div>
          <span>{`Message: ${message}`}</span>
          <button type="button" onClick={hideMessageHandler}>Hide</button>
        </div>
      )}
    </Fragment>
  );
}

Message.propTypes = {
  showMessage: PropTypes.bool.isRequired,
  message: PropTypes.string.isRequired,
  hideMessageHandler: PropTypes.func.isRequired,
};
