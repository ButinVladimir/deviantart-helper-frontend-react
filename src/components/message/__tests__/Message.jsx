import React from 'react';
import { shallow } from 'enzyme';
import Message from '../Message';

describe('Message', () => {
  const message = 'Message text';
  const messageColor = 'info';

  it('renders correctly when message is shown', () => {
    const wrapper = shallow(
      <Message
        showMessage
        message={message}
        messageColor={messageColor}
        hideMessageHandler={jest.fn()}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when message is hidden', () => {
    const wrapper = shallow(
      <Message
        showMessage={false}
        message={message}
        messageColor={messageColor}
        hideMessageHandler={jest.fn()}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can hide message', () => {
    const hideMessageHandler = jest.fn();
    const wrapper = shallow(
      <Message
        showMessage
        message={message}
        messageColor={messageColor}
        hideMessageHandler={hideMessageHandler}
      />,
    );

    wrapper.find('Button').simulate('click');

    expect(hideMessageHandler).toHaveBeenCalled();
  });
});
