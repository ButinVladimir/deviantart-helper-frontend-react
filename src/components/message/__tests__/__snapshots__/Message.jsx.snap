// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders correctly when message is hidden 1`] = `<Fragment />`;

exports[`Message renders correctly when message is shown 1`] = `
<Fragment>
  <Section
    className=""
    renderAs="section"
    size={null}
    style={Object {}}
  >
    <Container
      breakpoint="fullhd"
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Notification
        className=""
        color="info"
        renderAs="div"
        style={Object {}}
      >
        <Button
          className=""
          color={null}
          disabled={false}
          fullwidth={false}
          inverted={false}
          isSelected={false}
          isStatic={false}
          loading={false}
          onClick={[MockFunction]}
          outlined={false}
          remove={true}
          renderAs="button"
          reset={false}
          rounded={false}
          size={null}
          state={null}
          style={Object {}}
          submit={false}
          text={false}
        >
          Hide
        </Button>
        Message text
      </Notification>
    </Container>
  </Section>
</Fragment>
`;
